### PR TITLE
feat(platform): explain permission dependencies for scoped tokens

### PR DESCRIPTION
**Pull request by tobice**
Dependencies are tricky and can confuse users. Let's have them explained in the docs.

Related Slack discussion [here](https://apify.slack.com/archives/C05V6DJRWPJ/p1725632208478389) and [here](https://apify.slack.com/archives/C05V6DJRWPJ/p1725371887164209).

Related issue: https://github.com/apify/apify-core/issues/17384

### CODE REVIEW

**Review by fnesveda — APPROVED** [review]
Cool!

**Review by drobnikj — APPROVED** [review]
👍

File: sources/platform/integrations/programming/api.md [hunk 1]
```diff
@@ -90,6 +90,25 @@ This is useful if you want to for example create a token that can dynamically cr
 
 :::
 
+### Permission dependencies
+
+Some permissions have dependencies, that means other permissions which need to be granted as well.
```
Comment (TC-MO) [inline sources/platform/integrations/programming/api.md:95]:
```suggestion
Some permissions require other permissions to be granted alongside them. These are called _permission dependencies_.
```

File: sources/platform/integrations/programming/api.md [hunk 2]
```diff
@@ -90,6 +90,25 @@ This is useful if you want to for example create a token that can dynamically cr
...
+
+Some of these dependencies are enforced already by the form. You will notice that if you for example try to grant the _Write_ permission to a dataset, the _Read_ permission will be automatically selected as well. In other words, you can only write to a dataset if you can also read from it.
+
+![The Write permission depends on Read for a dataset](../images/api-token-scoped-dependencies.png)
```
Comment (TC-MO) [inline sources/platform/integrations/programming/api.md:99]:
```suggestion
#### Automatic dependencies

The form enforces certain dependencies automatically. For example, when you grant the _Write_ permission for a dataset, the _Read_ permission is automatically selected. This ensures that you can write to a dataset if you can also read from it.

![The Write permission depends on Read for a dataset](../images/api-token-scoped-dependencies.png)
```

File: sources/platform/integrations/programming/api.md [hunk 3]
```diff
@@ -90,6 +90,25 @@ This is useful if you want to for example create a token that can dynamically cr
...
+However, other dependencies are more complicated, and **it is your responsibility that the token is set up correctly**. Specifically:
+
+- To create or update a Schedule, the token needs access not only to the Schedule itself, but also to the Actor or task that is being scheduled.
+- Similarly, to create or update a task, the token needs the additional permission to access the task's Actor itself.
+
+:::tip
+
+Let's say that you have an Actor and you want to programmatically create schedules for that Actor. Then you can create a token that has the account level _Create_ permission on schedules, but only the resource-specific _Run_ permission on the Actor. Such a token has exactly the permissions it needs, and nothing more.
+
+:::
```
Comment (TC-MO) [inline sources/platform/integrations/programming/api.md:110]:
```suggestion
#### Manual dependencies

Other dependencies are more complex and require manual configuration. It's your responsibility to set up the token correctly.

Consider the following scenarios:

- Creating or updating a Schedule: The token need access to both the Schedule and the associated Actor or task.
- Creating or updating a task: The token needs permission to access both the task and its Actor.

:::tip

To create a token with minimal permission, consider this example:

You want to programmatically create schedules for an Actor. Create a token with:

- Account-level _Create_ permission on Schedules
- Resource-specific _Run_ permission on the Actor

This token has the permissions it needs, nothing more.

:::
```

**Review by TC-MO — APPROVED** [review]
I've proposed some rewrites to add a bit more structure and be a bit more inline with Msoft Style Guide

But I'm open to discussion about all the changes made.

The added subheadings won't show up on the right-side TOC, otherwise the goal was to:

* Restructure the content with additional subheadings for clarity
* Add more active voice & direct language
* Minimalizing weasel words

File: sources/platform/integrations/programming/api.md (hunk 1 above)
Comment (tobice) [reply sources/platform/integrations/programming/api.md:95]: Looks good, nice wording :+1:

File: sources/platform/integrations/programming/api.md (hunk 2 above)
Comment (tobice) [reply sources/platform/integrations/programming/api.md:99]: Are you sure there won't be too much nesting with the extra header? :thinking:

File: sources/platform/integrations/programming/api.md (hunk 3 above)
Comment (tobice) [reply sources/platform/integrations/programming/api.md:110]: There is a like nested "colonning" :smile: 

First level: `To create a token with minimal permission, consider this example:`
Second level: `You want to programmatically create schedules for an Actor. Create a token with:`

TBH, the original compact variant looks better to me.

File: sources/platform/integrations/programming/api.md (hunk 2 above)
Comment (TC-MO) [reply sources/platform/integrations/programming/api.md:99]: The `Automatic dependencies & `Manual dependencies` do not show up on right hand ToC, while within prose itself I think they nicely break up the concepts into their own place

Comment (tobice) [reply sources/platform/integrations/programming/api.md:99]: Makes sense.

**Pull request merged**